engine/src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -46,8 +72,25 @@ trait DataAccess extends AutoCloseable {
 
   def upsertExecutionInfo(workflowId: WorkflowId,
                           callKey: BackendCallKey,
+                          keyValues: Map[String, Option[String]])
+                         (implicit ec: ExecutionContext): Future[Unit]
+
+  /*
+  TODO: Would love a fancier adapter instead of this overload importing the implicits already available in the caller!
+  Maybe everywhere we'll have our own DBIO-ish object.
+  - dataAccess.upsertWhatever(params).withoutRetry(implicit executionContext)
+  - dataAccess.upsertWhatever(params).withRetry(implicit actorSystem) <-- assumes retrying 10 times
+
+  Or maybe a larger refactoring will use our own actors. TBD
+   */
+  def upsertExecutionInfo(workflowId: WorkflowId,
+                          callKey: BackendCallKey,
                           keyValues: Map[String, Option[String]],
-                          actorSystem: ActorSystem): Future[Unit]
+                          actorSystem: ActorSystem): Future[Unit] = {
+    implicit val system = actorSystem
+    implicit val ec = actorSystem.dispatcher
+    withRetry(upsertExecutionInfo(workflowId, callKey, keyValues))
+  }
```

Comment:
Perhaps the non-retrying abstract version of that method should be made `protected` so callers don't mistakenly invoke that.